units/patch_intent.go

```diff
@@ -488,73 +486,58 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
-	}
-	var res []patch.VariantTasks
-	if failedOnly && !(previousPatch.Status == evergreen.PatchFailed) {
-		return res, nil
-	}
+func setTasksToPreviousFailed(patchDoc, previousPatch *patch.Patch, project *model.Project) error {
+	var failedTasks []string
 	for _, vt := range previousPatch.VariantsTasks {
 		tasksInProjectVariant := project.FindTasksForVariant(vt.Variant)
 		displayTasksInProjectVariant := project.FindDisplayTasksForVariant(vt.Variant)
-		var displayTasks []patch.DisplayTask
 		var tasks []string
-		if failedOnly {
-			tasks, displayTasks, err = getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant, displayTasksInProjectVariant, vt, previousPatch.Version)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			tasks, displayTasks = getPreviousTasksAndDisplayTasks(tasksInProjectVariant, displayTasksInProjectVariant, vt)
-		}
-		if len(tasks)+len(displayTasks) > 0 {
-			res = append(res, patch.VariantTasks{
-				Variant:      vt.Variant,
-				Tasks:        tasks,
-				DisplayTasks: displayTasks,
-			})
+		tasks, err := getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant, displayTasksInProjectVariant, vt, previousPatch.Version)
+		if err != nil {
+			return err
 		}
+		failedTasks = append(failedTasks, tasks...)
+	}
+
+	patchDoc.Tasks = failedTasks
+	return nil
+}
+
+func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patch, project *model.Project, failedOnly bool) (string, error) {
+	previousPatch, err := patch.FindOne(patch.MostRecentPatchByUserAndProject(j.user.Username(), project.Identifier))
+	if err != nil {
+		return "", errors.Wrap(err, "querying for most recent patch")
+	}
+	if previousPatch == nil {
+		return "", errors.Errorf("no previous patch available")
 	}
-	return res, nil
+
+	patchDoc.BuildVariants = previousPatch.BuildVariants
+
+	if failedOnly {
+		setTasksToPreviousFailed(patchDoc, previousPatch, project)
+	} else {
+		patchDoc.Tasks = previousPatch.Tasks
+	}
+
+	return previousPatch.Status, nil
 }
 
-func getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant []string, displayTasksInProjectVariant []string, vt patch.VariantTasks, version string) ([]string, []patch.DisplayTask, error) {
+func getPreviousFailedTasksAndDisplayTasks(tasksInProjectVariant []string, displayTasksInProjectVariant []string, vt patch.VariantTasks, version string) ([]string, error) {
 	failedTasks, err := task.FindAll(db.Query(task.FailedTasksByVersionAndBV(version, vt.Variant)))
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "error querying for failed tasks from previous patch")
+		return nil, errors.Wrap(err, "error querying for failed tasks from previous patch")
 	}
 	failedExecutionTasks := []string{}
-	failedDisplayTasks := []string{}
 	for _, failedTask := range failedTasks {
-		if failedTask.DisplayOnly {
-			failedDisplayTasks = append(failedDisplayTasks, failedTask.DisplayName)
-		} else {
+		if !failedTask.DisplayOnly {
```

Comment:
Im a little confused 
are we assuming all failed tasks that arent a display task to be an execution task?